### ISSUE TITLE
Error running Slither 0.7.1: ValueError: invalid literal for int() with base 10: '1.1e18'

### ISSUE THREAD

**Issue by jsmjsm** [crytic/slither#820]
I have tried Slither 0.7.0 & 0.7.1 on VSCode and Terminal , still hit this problem..

```
Python 3.9.2
Solc Version: 0.6.9+commit.3e3065ac.Darwin.appleclang
Slither: 0.7.1
```
--------
Here is the log :

```
⸻ Slither: Solidity static analysis framework by Trail of Bits ⸻
Using slither version: 0.7.1
Refreshing explorer...
Loaded 0 issues, displaying 0
⸻ Starting analysis ⸻
Error: Error in workspace "/Users/myname/Documents/-code/code.nosync/projectname/contract-exchange-main":
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/__main__.py", line 721, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/__main__.py", line 82, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/__main__.py", line 65, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/slither.py", line 102, in __init__
    self._parser.parse_contracts()
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/slitherSolc.py", line 384, in parse_contracts
    self._analyze_third_part(contracts_to_be_analyzed, libraries)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/slitherSolc.py", line 490, in _analyze_third_part
    self._analyze_variables_modifiers_functions(contract)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/slitherSolc.py", line 553, in _analyze_variables_modifiers_functions
    contract.analyze_state_variables()
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/declarations/contract.py", line 519, in analyze_state_variables
    var_parser.analyze(self)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/variables/variable_declaration.py", line 181, in analyze
    self._variable.type = parse_type(self._elem_to_parse, caller_context)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/solidity_types/type_parsing.py", line 312, in parse_type
    mappingTo = parse_type(t["valueType"], next_context)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/solidity_types/type_parsing.py", line 312, in parse_type
    mappingTo = parse_type(t["valueType"], next_context)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/solc_parsing/solidity_types/type_parsing.py", line 306, in parse_type
    return ArrayType(array_type, length)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/core/solidity_types/array_type.py", line 22, in __init__
    cf = ConstantFolding(length, "uint256")
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/constants_folding.py", line 26, in __init__
    super().__init__(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 30, in __init__
    self._visit_expression(self.expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 95, in _visit_expression
    self._post_visit(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 280, in _post_visit
    self._post_identifier(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/constants_folding.py", line 37, in _post_identifier
    cf = ConstantFolding(expr, self._type)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/constants_folding.py", line 26, in __init__
    super().__init__(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 30, in __init__
    self._visit_expression(self.expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 48, in _visit_expression
    self._visit_binary_operation(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 104, in _visit_binary_operation
    self._visit_expression(expression.expression_left)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 48, in _visit_expression
    self._visit_binary_operation(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 104, in _visit_binary_operation
    self._visit_expression(expression.expression_left)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 81, in _visit_expression
    self._visit_tuple_expression(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 152, in _visit_tuple_expression
    self._visit_expression(e)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 48, in _visit_expression
    self._visit_binary_operation(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 104, in _visit_binary_operation
    self._visit_expression(expression.expression_left)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 95, in _visit_expression
    self._post_visit(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/expression.py", line 280, in _post_visit
    self._post_identifier(expression)
  File "/usr/local/lib/python3.9/site-packages/slither_analyzer-0.7.1-py3.9.egg/slither/visitors/expression/constants_folding.py", line 39, in _post_identifier
    set_val(expression, int(expr.value))
ValueError: invalid literal for int() with base 10: '1.1e18'


⸻ Analysis: 0 succeeded, 1 failed, 0 skipped ⸻
Refreshing explorer...
Loaded 0 issues, displaying 0
```

**Comment by montyly** [comment]
Hi @jsmjsm 

Thanks for reporting this.  

It seems to be happening in our constant propagation if the constant is used later as the size of an array. Minimal testcase:
```solidity
contract C{
    uint constant test = 1.1e18;
    uint[test] arrays;
}
```

**Comment by jsmjsm** [comment]
> Hi @jsmjsm
> 
> Thanks for reporting this.
> 
> It seems to be happening in our constant propagation if the constant is used later as the size of an array. Minimal testcase:
> 
> ```solidity
> contract C{
>     uint constant test = 1.1e18;
>     uint[test] arrays;
> }
> ```

I have checked my code, there is no code written in this style.

**Comment by jsmjsm** [comment]
🎆 I changed the value of constant the  `1.1e18` to `1.1*10**18` and the problem has been solved.

**Comment by montyly** [comment]
Hi @jsmjsm. We fixed the issue in #830, it will be available in the upcoming release (#829)
